website/content/en/preview/AWS/user-data.md

```diff
@@ -0,0 +1,129 @@
+---
+title: "User Data Configuration"
+linkTitle: "UserData"
+weight: 10
+description: >
+  Learn how to configure custom UserData with Karpenter
+---
+
+This document describes how you can customize the UserData that will be specified on your EC2 worker nodes, without using a launch template.
+
+## Configuration
+
+In order to specify custom user data, you must include it within a AWSNodeTemplate resource. You can then reference this AWSNodeTemplate resource through `spec.providerRef` in your provisioner.
+
+**Examples**
+
+Your UserData can be added to `spec.userData` in the `AWSNodeTemplate` resource like this -
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: mynodetemplate
+spec:
+  userData:  |
+    [settings.kubernetes]
+    kube-api-qps = 30
+    [settings.kubernetes.eviction-hard]
+    "memory.available" = "20%"
+```
+
+The AWSNodeTemplate CRD can then be referenced within the provisioner through `providerRef` -
+```
+spec:
+  provider:
+    amiFamily: Bottlerocket
```

Comment:
From our discussion, I thought provider and providerRef was an OR semantic. The user would need to specify these in in the AWSNodeTemplate.

Reply:
That's a much larger change to make. I can work on it in a separate PR - do you think we should disable this feature until that's complete?

Reply:
Yeah probably.